src/plugins/dashboard/public/dashboard_app/no_data/dashboard_app_no_data.tsx

```diff
@@ -44,6 +58,63 @@ export const DashboardAppNoDataPage = ({
       })
     )
   );
+
+  const lensHelpersAsync = useAsync(() => {
+    return lensService?.stateHelperApi() ?? Promise.resolve(null);
+  }, [lensService]);
+
+  const onTryESQL = useCallback(async () => {
+    if (lensHelpersAsync.value) {
+      const { dataViews } = dataService;
+      const indexName = (await getIndexForESQLQuery({ dataViews })) ?? '*';
+      const dataView = await getESQLAdHocDataview(`from ${indexName}`, dataViews);
+      const esqlQuery = getInitialESQLQuery(dataView);
+      const abortController = new AbortController();
+
+      const columns = await getESQLQueryColumns({
+        esqlQuery,
+        search: dataService.search.search,
+        signal: abortController.signal,
+        timeRange: dataService.query.timefilter.timefilter.getAbsoluteTime(),
+      });
+
+      // lens suggestions api context
+      const context = {
```

Comment:
You should add a check that component is still mounted after async actions. Otherwise, users can navigate to a different page while async stuff is happening and then get whisked back to a new esql dashboard when the rest of the function executes after async code completes. Also, on unmount, abortController should be aborted.

Reply:
Done https://github.com/elastic/kibana/pull/196658/commits/f806040487bcc820e8680b3e4c4af369ae5fe0c0